networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/topics/topichandlers/AggregateAttestationTopicHandler.java

```diff
@@ -13,43 +13,33 @@
 
 package tech.pegasys.teku.networking.eth2.gossip.topics.topichandlers;
 
-import org.apache.tuweni.bytes.Bytes;
 import tech.pegasys.teku.datastructures.attestation.ValidateableAttestation;
 import tech.pegasys.teku.datastructures.operations.SignedAggregateAndProof;
 import tech.pegasys.teku.infrastructure.async.AsyncRunner;
-import tech.pegasys.teku.networking.eth2.gossip.encoding.DecodingException;
 import tech.pegasys.teku.networking.eth2.gossip.encoding.GossipEncoding;
 import tech.pegasys.teku.networking.eth2.gossip.topics.OperationProcessor;
-import tech.pegasys.teku.networking.p2p.gossip.PreparedGossipMessage;
 import tech.pegasys.teku.ssz.SSZTypes.Bytes4;
 
-public class AggregateAttestationTopicHandler extends Eth2TopicHandler<ValidateableAttestation> {
+public class AggregateAttestationTopicHandler {
+
   public static String TOPIC_NAME = "beacon_aggregate_and_proof";
 
-  public AggregateAttestationTopicHandler(
+  public static Eth2TopicHandler<?> createHandler(
       final AsyncRunner asyncRunner,
       final OperationProcessor<ValidateableAttestation> operationProcessor,
       final GossipEncoding gossipEncoding,
       final Bytes4 forkDigest) {
-    super(
+
+    OperationProcessor<SignedAggregateAndProof> convertingProcessor =
+        proofMessage ->
+            operationProcessor.process(
+                ValidateableAttestation.aggregateFromValidator(proofMessage));
```

Comment:
Should this be:
```suggestion
                ValidateableAttestation.aggregateFromNetwork(proofMessage));
```
?

Reply:
Hmm, you seem to be right
Will make a separate PR for this, since `aggregateFromValidator()` was here before refactoring

Reply:
https://github.com/ConsenSys/teku/pull/3544